src/AcceptanceTests/Receiving/When_receiving_a_message.cs

```diff
@@ -0,0 +1,67 @@
+﻿namespace NServiceBus.Transport.AzureServiceBus.AcceptanceTests.Sending.Receiving
+{
+    using System;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using NServiceBus.AcceptanceTests.EndpointTemplates;
+    using NUnit.Framework;
+    using Pipeline;
+
+    public class When_receiving_a_message
+    {
+        [Test]
+        public async Task Should_have_access_to_the_original_brokered_message_via_the_context_bag()
+        {
+            await Scenario.Define<Context>()
+                .WithEndpoint<Endpoint>(b => b.When(
+                    (session, c) => session.SendLocal(new Message())))
+                .Done(c => c.OriginalMessageFound)
+                .Run();
+        }
+
+        public class Context : ScenarioContext
+        {
+            public bool OriginalMessageFound { get; set; }
+        }
+
+        public class Endpoint : EndpointConfigurationBuilder
+        {
+            public Endpoint()
+            {
+                EndpointSetup<DefaultServer>((c, d) =>
+                    c.Pipeline.Register(b => new CheckContextForValidUntilUtc(d.ScenarioContext as Context), "Behavior to validate context bag contains the original brokered message"));
+            }
+
+            public class Handler : IHandleMessages<Message>
+            {
+                public Context TestContext { get; set; }
+
+                public Task Handle(Message request, IMessageHandlerContext context)
+                {
+                    TestContext.OriginalMessageFound = TestContext.OriginalMessageFound && context.Extensions.Get<Microsoft.Azure.ServiceBus.Message>() != null;
+
+                    return Task.CompletedTask;
+                }
+            }
+
+            public class CheckContextForValidUntilUtc : Behavior<ITransportReceiveContext>
+            {
+                readonly Context testContext;
+
+                public CheckContextForValidUntilUtc(Context context)
+                {
+                    testContext = context;
+                }
+
+                public override Task Invoke(ITransportReceiveContext context, Func<Task> next)
+                {
+                    testContext.OriginalMessageFound = context.Extensions.Get<Microsoft.Azure.ServiceBus.Message>() != null;
```

Comment:
Should this use ` GetOrDefault` ? if the message is not here this would throw an exception and then cause the test to timeout right?

Reply:
There's GetOrCreate(), but I'd rather keep it as Get. If the message is not found, the test will fail.